# Ab initio Gene Prediction Softwares List:

- **Eukaryotes:**
     
| Year | Tool Name | Publication | Method | Organism |
| ---- | --------- | ----------- | ------ | -------- |
| 1994 | GeneLang | Dong, S., & Searls, D. B. (1994). Gene structure prediction by linguistic methods. Genomics, 23(3), 540–551. https://doi.org/10.1006/geno.1994.1541.<br> | Linguistic method, Hidden Markov Model, Dynamic Programming, Weight Array Matrix | Eukaryote |
| 1995 | Fgeneh / GeneFinder | Solovyev, V. V., Salamov, A. A., & Lawrence, C. B. (1995). Identification of human gene structure using linear discriminant functions and dynamic programming. Proceedings. International Conference on Intelligent Systems for Molecular Biology, 3, 367–375.<br> | Hidden Markov Model, Dynamic Programming, Linear Discriminant Analysis | Human |
| 1997 | HMMGene | Krogh, A. (1997). Two methods for improving performance of an HMM and their application for gene finding. Proceedings. International Conference on Intelligent Systems for Molecular Biology, 5, 179–186.<br> | Hidden Markov Model Conditional Random Field | Vertebrate and C. elegans |
| 1997 | GenScan | Burge, C., & Karlin, S. (1997). Prediction of complete gene structures in human genomic DNA. Journal of Molecular Biology, 268(1), 78–94. https://doi.org/10.1006/jmbi.1997.0951.<br> | Generalized Hidden Markov Model ||
| 2003 | AUGUSTUS | Stanke, M., & Waack, S. (2003). Gene prediction with a hidden Markov model and a new intron submodel. Bioinformatics (Oxford, England), 19 Suppl 2, ii215-225. https://doi.org/10.1093/bioinformatics/btg1080<br> | Hidden Markov Model | Eukaryote |
| 2004 | GlimmerHMM | Majoros, W. H., Pertea, M., & Salzberg, S. L. (2004). TigrScan and GlimmerHMM: Two open source ab initio eukaryotic gene-finders. Bioinformatics (Oxford, England), 20(16), 2878–2879. https://doi.org/10.1093/bioinformatics/bth315<br> | Generalized Hidden Markov Model | Eukaryote |
| 2005 | GeneMark-ES | Lomsadze, A., Ter-Hovhannisyan, V., Chernoff, Y. O., & Borodovsky, M. (2005). Gene identification in novel eukaryotic genomes by self-training algorithm. Nucleic Acids Research, 33(20), 6494–6506. https://doi.org/10.1093/nar/gki937.<br> | | Eukaryote |
| 2005 | BGF (Beijing Gene Finder) | Li, H., Liu, J.-S., Xu, Z., Jin, J., Fang, L., Gao, L., Li, Y.-D., Xing, Z.-X., Gao, S.-G., Liu, T., Li, H.-H., Li, Y., Fang, L.-J., Xie, H.-M., Zheng, W.-M., & Hao, B.-L. (2005). Test Data Sets and Evaluation of Gene Prediction Programs on the Rice Genome. Journal of Computer Science and Technology, 20(4), 446–453. https://doi.org/10.1007/s11390-005-0446-x.<br> | Semi Hidden Markov Model | Eukaryote |
| 2010 | Gnomon | Souvorov, A., Kapustin, Y., Kiryutin, B., Chetvernin, V., Tatusova, T., & Lipman, D. (2010). Gnomon – NCBI eukaryotic gene prediction tool. Available in: <https://www.ncbi.nlm.nih.gov/core/assets/genome/files/Gnomon-description.pdf> <br>| Hidden Markov Model, Weight Array Matrix| |
| 2014 | GeneMark-ET | Lomsadze, A., Burns, P. D., & Borodovsky, M. (2014). Integration of mapped RNA-Seq reads into automatic training of eukaryotic gene finding algorithm. Nucleic Acids Research, 42(15), e119. https://doi.org/10.1093/nar/gku557.<br> | Hidden Markov Model | Eukaryote |

---

- **Prokaryotes:**

| Year | Tool Name | Publication | Method | Organism |
| ---- | --------- | ----------- | ------ | -------- |
| 1998 | GeneMark.hmm | Lukashin, A. V., & Borodovsky, M. (1998). GeneMark.hmm: New solutions for gene finding. Nucleic Acids Research, 26(4), 1107–1115. https://doi.org/10.1093/nar/26.4.1107.<br> | Hidden Markov Model | Prokaryote / Archaea |
| 2001 | GeneHacker Plus | Yada, T., Totoki, Y., Takagi, T., & Nakai, K. (2001). A novel bacterial gene-finding system with improved accuracy in locating start codons. DNA Research: An International Journal for Rapid Publication of Reports on Genes and Genomes, 8(3), 97–106. https://doi.org/10.1093/dnares/8.3.97.<br> | Hidden Markov Model | Prokaryote |
| 2001 | GeneMarkS | Besemer, J., Lomsadze, A., & Borodovsky, M. (2001). GeneMarkS: A self-training method for prediction of gene starts in microbial genomes. Implications for finding sequence motifs in regulatory regions. Nucleic Acids Research, 29(12), 2607–2618. https://doi.org/10.1093/nar/29.12.2607.<br> | Hidden Markov Model | Prokaryote |
| 2003 | AMIGene | Bocs, S., Cruveiller, S., Vallenet, D., Nuel, G., & Médigue, C. (2003). AMIGene: Annotation of MIcrobial Genes. Nucleic Acids Research, 31(13), 3723–3726.<br> | Hidden Markov Model | Prokaryote |
| 2003 | EasyGene | Larsen, T. S., & Krogh, A. (2003). EasyGene – a prokaryotic gene finder that ranks ORFs by statistical significance. BMC Bioinformatics, 4(1), 21. https://doi.org/10.1186/1471-2105-4-21.<br> | Hidden Markov Model | Prokaryote / Archaea |
| 2003 | ZCurve | Guo, F.-B., Ou, H.-Y., & Zhang, C.-T. (2003). ZCURVE: A new system for recognizing protein-coding genes in bacterial and archaeal genomes. Nucleic Acids Research, 31(6), 1780–1789. https://doi.org/10.1093/nar/gkg254.<br> | Z-curve | Prokaryote / Archaea |
| 2006 | MetaGeneAnnotator (MGA) | Noguchi, H., Taniguchi, T., & Itoh, T. (2008). MetaGeneAnnotator: Detecting Species-Specific Patterns of Ribosomal Binding Site for Precise Gene Prediction in Anonymous Prokaryotic and Phage Genomes. DNA Research, 15(6), 387–396. https://doi.org/10.1093/dnares/dsn027.<br> | | Prokaryote |
| 2007 | GISMO | Krause, L., McHardy, A. C., Nattkemper, T. W., Pühler, A., Stoye, J., & Meyer, F. (2007). GISMO--gene identification using a support vector machine for ORF classification. Nucleic Acids Research, 35(2), 540–549. https://doi.org/10.1093/nar/gkl1083.<br> | Support Vector Machine | Prokaryote |
| 2010 | Prodigal (PROkaryotic DYnamic programming Gene-finding ALgorithm) | Hyatt, D., Chen, G.-L., LoCascio, P. F., Land, M. L., Larimer, F. W., & Hauser, L. J. (2010). Prodigal: Prokaryotic gene recognition and translation initiation site identification. BMC Bioinformatics, 11(1), 119. https://doi.org/10.1186/1471-2105-11-119. <br>| Dynamic Programming + Hidden Markov Model | Prokaryote |
| 2014 | ZUPLS | Song, K., Tong, T., & Wu, F. (2014). Predicting essential genes in prokaryotic genomes using a linear method: ZUPLS. Integrative Biology: Quantitative Biosciences from Nano to Macro, 6(4), 460–469. https://doi.org/10.1039/c3ib40241j.<br> | Z-curve | Prokaryote |

---

- **Virus:**

| Year | Tool Name | Publication | Method | Organism |
| ---- | --------- | ----------- | ------ | -------- |
| 2003 | GeneMarkS (virus version) | Mills, R., Rozanov, M., Lomsadze, A., Tatusova, T., & Borodovsky, M. (2003). Improving gene annotation of complete viral genomes. Nucleic Acids Research, 31(23), 7041–7055. https://doi.org/10.1093/nar/gkg878.<br>  | Hidden Markov Model | Virus |
| 2006 | ZCURVE_V | Guo, F.-B., & Zhang, C.-T. (2006). ZCURVE_V: A new self-training system for recognizing protein-coding genes in viral and phage genomes. BMC Bioinformatics, 7(1), 9. https://doi.org/10.1186/1471-2105-7-9. <br>  | Z-curve | Virus |
| 2007 | GLIMMER3 | Delcher, A. L., Bratke, K. A., Powers, E. C., & Salzberg, S. L. (2007). Identifying bacterial genes and endosymbiont DNA with Glimmer. Bioinformatics, 23(6), 673–679. https://doi.org/10.1093/bioinformatics/btm009. <br>  | Interpolated Markov Model | Bacteria, Archaea, and Viruses |
| 2019 | Vgas (Viral Genome Annotation System) | Zhang, K.-Y., Gao, Y.-Z., Du, M.-Z., Liu, S., Dong, C., & Guo, F.-B. (2019). Vgas: A Viral Genome Annotation System. Frontiers in Microbiology, 10. https://doi.org/10.3389/fmicb.2019.00184.<br>  | ZCURVE_V + BLASTp | Virus |

---

- **Metagenome:**

| Year | Tool Name | Publication | Method | Organism |
| ---- | --------- | ----------- | ------ | -------- |
| 2006 | Metagene | Noguchi, H., Park, J., & Takagi, T. (2006). MetaGene: Prokaryotic gene finding from environmental genome shotgun sequences. Nucleic Acids Research, 34(19), 5623–5630. https://doi.org/10.1093/nar/gkl723.<br>  | | Metagenome |
| 2010 | FragGeneScan | Rho, M., Tang, H., & Ye, Y. (2010). FragGeneScan: Predicting genes in short and error-prone reads. Nucleic Acids Research, 38(20), e191. https://doi.org/10.1093/nar/gkq747. <br>  | Hidden Markov Model | Metagenome |
| 2010 | MetaGeneMark | Zhu, W., Lomsadze, A., & Borodovsky, M. (2010). Ab initio gene identification in metagenomic sequences. Nucleic Acids Research, 38(12), e132. https://doi.org/10.1093/nar/gkq275.<br>  | Hidden Markov Model | Metagenome |
| 2013 | MetaGUN | Liu, Y., Guo, J., Hu, G., & Zhu, H. (2013). Gene prediction in metagenomic fragments based on the SVM algorithm. BMC Bioinformatics, 14(5), S12. https://doi.org/10.1186/1471-2105-14-S5-S12.<br>  | Support Vector Machine | Metagenome |